src/libcore/thread.cpp

```diff
@@ -85,12 +88,30 @@ class WorkerThread : public Thread {
 
 std::atomic<uint32_t> WorkerThread::m_counter{0};
 
+struct ThreadNotifier {
+    ThreadNotifier() {
+        // Do not register the main thread
+        if (m_counter > 0)
+            Thread::register_external_thread("wrk_");
+        m_counter++;
+    }
+    ~ThreadNotifier() {
+        if (self)
+            Thread::unregister_external_thread();
+        m_counter--;
+    }
+    static std::atomic<uint32_t> m_counter;
+}
+
+static thread_local ThreadNotifier;
```

Comment:
I am confused by this line. TLS behavior is largely plaform-dependent. As such, I don't think we can expect that this will be correctly initialized everywhere in such an anonymous variable-- rather, some kind of access to this variable will be needed, which will then check if the object has been constructed. If you know more than I do, then I'd be curious to learn more.

Reply:
Would you like this better?
```
static thread_local ThreadNotifier notifier{};
```

Reply:
No, I don't think that really addresses the issue in a cross-platform way. The problem is that `notifier` will not automatically get called if a new thread launches (it seems to me you are assuming that happens everywhere, correct?)